eng/common/pipelines/templates/steps/verify-links.yml

```
@@ -1,19 +1,21 @@
 parameters:
   Directory: 'not-specified'
   IgnoreLinksFile: "$(Build.SourcesDirectory)/eng/ignore-links.txt"
-  
+  WorkingDirectory: $(System.DefaultWorkingDirectory)
+  ScriptDirectory: eng/common/scripts
 steps:
-- task: PowerShell@2
-  displayName: Link verification check
-  inputs:
-    pwsh: true
-    workingDirectory: $(Build.SourcesDirectory)/${{ parameters.Directory }}
-    filePath: eng/common/scripts/Verify-Links.ps1 
-    arguments: >
-      -urls $(dir -r -i *.md) 
-      -rootUrl "file://$(Build.SourcesDirectory)/${{ parameters.Directory }}" 
-      -recursive: $false 
-      -ignoreLinksFile ${{ parameters.IgnoreLinksFile }}
-      -branchReplaceRegex "^($env:SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI.*/(?:blob|tree)/)master(/.*)$"
-      -branchReplacementName $env:SYSTEM_PULLREQUEST_SOURCECOMMITID
-      -devOpsLogging: $true
+  - task: PowerShell@2
+    displayName: Link verification check
+    inputs:
+      pwsh: true
+      workingDirectory: "${{ parameters.WorkingDirectory }}/${{ parameters.Directory }}"
```

I don't think this should contain directory. 
```suggestion
      workingDirectory: "${{ parameters.WorkingDirectory }}"
```

---

The script is running against all the markdown under working directory, so we need to take in the $serviceDirectory for workingDirectory, otherwise we will run the entire repo e.g. /azure-sdk-for-java

---

OK yes the dir is against the current working directory. However now we have a chance that the rooturl and the working directory are not in sync. We need to make those consistent.